app/controllers/reviews_controller.rb

```diff
@@ -2,12 +2,12 @@ class ReviewsController < ApplicationController
   before_action :authenticate_user!, only: [:update, :edit, :destroy]
 
   def new
-    @piece = Piece.find(params[:piece_id])
+    piece
```

Comment:
Remove this and create a controller callback `before_action :piece, only: [:new, :create, :update]`